### PR TITLE
Feature/tugcan/ui improvements

### PR DESCRIPTION
**Pull request by tugcanolgun**
* Add remove dependencies from head
* Add many functions, change source code of onepagescroll
* Change styling from custom to bootstrap with touch of custom style
* Add index page, Remove restriction from security (needs to reverted)
* Add index page, put map with 10 random location markers
* Add location, annotation sections
* Change layout for login add bootstrap
* Change layout for register and add error
* Add header footer and links to the welcome screen
* Fix location list, added fontawesome icons for added locations

I have changed onepagescroll's source-code so this local one is the one we must use.
Annotations section is currently empty
Add memory page is again on the whitelist right now for easy demonstration in the class. I have that in my mind.
I have tried non-strict version of thymeleaf and changed dependencies and such but unfortunately if we don't want to go back to 90's, we have to stick with this dictator-alike XML parser. May cause problems in the future due to quilljs's parsing capabilities.

### CODE REVIEW

File: mnemosyne/src/main/java/com/boun/swe/mnemosyne/service/MemoryService.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 import java.util.List;
 
-import static com.sun.tools.doclets.formats.html.markup.HtmlStyle.title;
+//import static com.sun.tools.doclets.formats.html.markup.HtmlStyle.title;
```
Comment (vedatakses) [inline mnemosyne/src/main/java/com/boun/swe/mnemosyne/service/MemoryService.java]: Let's remove this line instead

File: mnemosyne/src/main/java/com/boun/swe/mnemosyne/service/MemoryService.java [hunk 2]
```diff
@@ -27,7 +27,6 @@ public MemoryService(MemoryRepository memoryRepository) {
 
     public Memory createMemory(Memory memory) {
         final Memory storedMemory = memoryRepository.save(memory);
-        LOGGER.info("Memory with title: {} created successfully", title);
```
Comment (vedatakses) [inline mnemosyne/src/main/java/com/boun/swe/mnemosyne/service/MemoryService.java:30]: We can use this one.. 

`LOGGER.info("Memory with title: {} created successfully", memory.getTitle());`

File: mnemosyne/src/main/java/com/boun/swe/mnemosyne/service/MemoryService.java [hunk 3]
```diff
@@ -43,7 +42,6 @@ public Memory updateMemory(Memory memory) {
             throw new MemoryNotFoundException("Unable to find memory with id: " + memory.getId());
         }
         final Memory storedMemory = memoryRepository.save(memory);
-        LOGGER.info("Memory with title: {} updated successfully", title);
```
Comment (vedatakses) [inline mnemosyne/src/main/java/com/boun/swe/mnemosyne/service/MemoryService.java:46]: same above, mistakenly title used from silly library:

`LOGGER.info("Memory with title: {} updated successfully", memory.getTitle());`

**Review by vedatakses — APPROVED** [review]
Looks great man!

**Comment by tugcanolgun** [comment]
Merged into develop after fixes.

**Pull request merged**